src/components/ProductCard/ProductCard.jsx

```diff
@@ -1,18 +1,67 @@
-import { AiFillStar } from "../../icons/icons";
+import { useState } from "react";
+import { useCartContext, useWishlistContext ,useToastContext} from "../../context/context";
+import { AiFillStar, FaArrowRight, FaHeart } from "../../icons/icons";
+import { Loading } from "../components";
+import { Link } from "react-router-dom";
+
 import "./ProductCard.css";
-function ProductCard({ productCardDetails, btnTxt, productCardIcon }) {
+function ProductCard({ productCardDetails, btnTxt }) {
+  const { Cart, addToCart } = useCartContext();
+  const {Toast} = useToastContext();
+  const { Wishlist, addToWishlist, removeFromWishlist } = useWishlistContext();
+  const [wishlistLoader, showWishlistLoader] = useState(false);
+  const [isAddingToCart, setIsAddingToCart] = useState(false);
+  
   const ratingsArray = [];
   for (let i = 1; i <= productCardDetails.starRating; i++) {
     ratingsArray.push({
       icon: <AiFillStar key={i} className="size-xs icon" />,
     });
   }
 
+
+  function isPresentInState(data, state) {
```

Comment:
the same function you can import here to use if you extract it

Reply:
Okay Made the changes in next Pr.